dev/benchmarks/macrobenchmarks/lib/src/web/bench_mouse_region_grid_hover.dart

```diff
@@ -0,0 +1,153 @@
+import 'package:flutter/scheduler.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'recorder.dart';
+
+class _NestedMouseRegion extends StatelessWidget {
+  const _NestedMouseRegion({this.nests, this.child});
+
+  final int nests;
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    Widget current = child;
+    for (int i = 0; i < nests; i++) {
+      current = MouseRegion(
+        onEnter: (_) {},
+        child: child,
+      );
+    }
+    return current;
+  }
+}
+
+/// Creates a grid of mouse regions, then continuously hover over them.
+///
+/// Measures our ability to hit test mouse regions.
+class BenchMouseRegionGridHover extends WidgetRecorder {
+  BenchMouseRegionGridHover() : super(name: benchmarkName);
+
+  static const String benchmarkName = 'bench_mouse_region_grid_hover';
+
+  final _Tester tester = _Tester();
+
+  // Use a non-trivial border to force Web to switch painter
+  Border _getBorder(int columnIndex, int rowIndex) {
+    const BorderSide defaultBorderSide = BorderSide();
+
+    return Border(
+      left: columnIndex == 0 ? defaultBorderSide : BorderSide.none,
+      top: rowIndex == 0 ? defaultBorderSide : BorderSide.none,
+      right: defaultBorderSide,
+      bottom: defaultBorderSide,
+    );
+  }
+
+  bool started = false;
+
+  @override
+  void frameDidDraw() {
+    if (!started) {
+      started = true;
+      SchedulerBinding.instance.addPostFrameCallback((Duration timeStamp) async {
+        tester.start();
+      });
+    }
+    super.frameDidDraw();
+  }
+
+  @override
+  Widget createWidget() {
+    const int rowsCount = 60;
+    const int columnsCount = 20;
+    const double containerSize = 20;
+    return Directionality(
+      textDirection: TextDirection.ltr,
+      child: Align(
+        alignment: Alignment.topLeft,
+        child: SizedBox(
+          width: 400,
+          height: 400,
+          child: ListView.builder(
+            itemCount: rowsCount,
+            cacheExtent: rowsCount * containerSize,
+            physics: const ClampingScrollPhysics(),
+            itemBuilder: (BuildContext context, int rowIndex) => _NestedMouseRegion(
+              nests: 10,
+              child: Row(
+                children: List<Widget>.generate(
+                  columnsCount,
+                  (int columnIndex) => _NestedMouseRegion(
+                    nests: 10,
+                    child: Container(
+                      decoration: BoxDecoration(
+                        border: _getBorder(columnIndex, rowIndex),
+                        color: Color.fromARGB(255, rowIndex * 20 % 256, 127, 127),
+                      ),
+                      width: containerSize,
+                      height: containerSize,
+                    ),
+                  ),
+                ),
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class _Tester {
+  static const Duration hoverDuration = Duration(milliseconds: 20);
```

Comment:
That's barely over 1 frame. Is that long enough? It looks like it will be jumping between the 4 corners way too quickly, no?

Reply:
This duration is actually only used for the `timeStamp` of the event. The actual events are dispatched as fast as possible. The idea is to test the execution time the hit testing function.